docs/guides/js-waku/use-waku-react.md

```diff
@@ -0,0 +1,356 @@
+
+Use the `useCreateContentPair()` hook to create a message `encoder` and `decoder` pair:
+
+```js title="App.js"
+import { useCreateContentPair } from "@waku/react";
+
+function App() {
+	// Choose a content topic
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const ephemeral = false;
+
+	// Create a message encoder and decoder pair
+	const { encoder, decoder } = useCreateContentPair(contentTopic, ephemeral);
+
+	// "encoder" is the message encoder
+	// "decoder" is the message decoder
+	// Use these to handle the messages in your application
+}
+```
+
+## Send Messages Using Light Push
+
+Use the `useLightPush()` hook to bind `Light Push` methods to a node and `encoder`:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useLightPush,
+} from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+import { utf8ToBytes } from "@waku/sdk";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node, error, isLoading } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create an encoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { encoder  } = useCreateContentPair(contentTopic);
+
+	// Wait for the node to finish loading before sending messages
+	// (isLoading === false)
+
+	// Bind push method to a node and encoder
+	const { push } = useLightPush({ node, encoder });
+
+	// Send the message using Light Push
+	const sendMessage = async (text) => {
+		if (!push || !text) {
+			return;
+		}
+		const payload = utf8ToBytes(text);
+		await push({ payload });
+	};
+	sendMessage("Hello, World!");
+}
+```
+
+:::info
+Wait for the node to finish loading before sending messages (`isLoading` === `false`).
+:::
+
+## Receive Messages Using Filter
+
+Use the `useFilterMessages()` hook to receive messages from a `Filter` subscription:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useFilterMessages,
+} from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Receive messages from Filter subscription
+	const { error, messages, isLoading } = useFilterMessages({ node, decoder });
+	console.log(messages);
+
+	// "error" captures any error that occurs while receiving messages
+	// "messages" contains a list of messages the subscription received
+	// "isLoading" indicates whether the node is still subscribing to Filter
+	// Use these to handle states and render the UI in your application
+}
+```
+
```

Comment:
In the website there is an info block "wait for the messages to finish loading before processing them..".

I am not sure this is very clear. Let's improve the code first to display messages in a component and also have it wait for node to be loaded before it calls `useFilterMessages`.